source/blog/2018-05-30-emberjs-accessibility-and-a11y-tools.md

```diff
@@ -0,0 +1,153 @@
+
+Since the errors have ratings, we know to start with the critical error, which is that a form has missing labels. Screen readers rely on labels, not placeholder text, to help users navigate forms. Also, it's better UX for everyone.
+
+Here's what the form looks like before changes:
+
+```handlebars
+{{input id="filterByCity" value=value key-up=(action 'handleFilterEntry') class="light" placeholder="Filter By City"}}
+```
+
+![A form without labels](/images/blog/2018-05-30-emberjs-accessibility-and-a11y-tools/form-without-labels.png)
+
+Let's add the label. The `for` attribute should match the input id.
+
+```handlebars
+<label for="filterByCity" class="screen-reader">
+    Filter By City
+</label>
+
+{{input id="filterByCity" value=value key-up=(action 'handleFilterEntry') class="light" placeholder="Filter By City"}}
+```
+
+![A form with labels](/images/blog/2018-05-30-emberjs-accessibility-and-a11y-tools/form-with-labels.png)
+
+Now the filter has a label, and the tests pass!
+
+### Using ember-a11y-landmarks for roles
+
+The [ember-a11y-landmarks](https://github.com/ember-a11y/ember-a11y-landmarks) addon helps manage the roles that should go on semantic tags like `<header>` and `<nav>`, without needing to learn the intricacies of when to use them.
+
+One huge thing you can do to improve an app's accessibility is to use semantic tags in the HTML. For example, always use `<button>` for buttons, instead of making them `divs`. However, some semantic tags still need extra attributes called roles for screen readers to work right, and they only need them some of the time based on their position in the DOM. ember-a11y-landmarks to the rescue!
+
+To use this addon, wrap content in the ally-landmark block and say what kind of tag to use. Here's a template example:
+
+```handlebars
+{{#a11y-landmark tagName="footer"}}
+   This is a footer!
+{{/a11y-landmark}}
+And here's the html output with the correct role assigned:
+<footer id="ember337" role="contentinfo" class="ember-view"> 
+  This is a footer!
+</footer>
+```
+
+## Structural application concerns
+
+I learned from [Robert DeLuca](https://github.com/Robdel12) that [most front end frameworks have flaws](https://medium.com/@robdel12/single-page-apps-routers-are-broken-255daa310cf) that affect Assistive Technology like screen readers. However, there's a lot of progress being made in Ember itself, our community addons, and by our pals working in other frameworks! You can learn more in the recent JavaScript [State of Accessibility](https://www.youtube.com/watch?v=aoyLG2gTFpI) panel hosted by [ThisDot Media](https://www.thisdot.co/). Jump to the Ember segment presented by Melanie Sumner [here](https://youtu.be/aoyLG2gTFpI?t=2432).
+
+With the release of 3.1, Ember.js introduced some new features within the framework itself, and community addons continue to help drive progress on the remaining concerns. Historically, many new features in Ember originated as community addons.
```

Comment:
The primary concern of Robert's article seems to be the route-change-announcement issue, which is at least addressable in Ember with the `ember-a11y` as you mention a couple of sections below. If the reader of this blog post doesn't read Robert's article, I think this section makes sense because the reader will infer the scope of that article as being quite large. If they do read Robert's article, then come back here, they might be a little confused because the scope is pretty narrow.

In my opinion, the article can be more directly tied to the "Using the ember-a11y addon for focus management" section because of the article's focus. It could also be mentioned in the "What next?" section as something someone could advocate for being the default.

Reply:
Agreed, it should be moved. I rolled it into "Using the ember-a11y addon for focus management"